mlir/lib/Dialect/MIOpen/Transforms/ThreadwiseGemmLowering.cpp

```diff
@@ -194,136 +202,163 @@ struct XdlopsGemmV2RewritePattern : public OpConversionPattern<XdlopsGemmV2Op> {
+            loc, argType, matrixAElement, innerLoopiv);
+        Value argB = innerLoopb.create<ExtractSliceOp>(
+            loc, argType, matrixBElement, innerLoopiv);
+        populateMfma(innerLoopb, argA, argB, regCOffset);
       }
-    } else {
-      // for(index_t k_i = 0; k_i < KPerThread; ++k_i) {
-      //   matrixAElement = a[k_i];
-      //   matrixBElement = b[k_i];
-      //   // Loop within a kpack
-      //   for(index_t ki_i = 0; ki_i < k_base * KRepeats; ki_i += k_base)
-      //     argA = &matrixAElement[ki_i];
-      //     argB = &matrixAElement[ki_i];
-      //     p_c_thread = mfma_type.template run<MPerXlops * MRepeats,
-      //                                         NPerXdlops * NRepeats,
-      //                                         AStride,
-      //                                         BStride>(argA, argB,
-      //       p_c_thread);
-      // }
-      int64_t outerLoopUpperBound = KPerThread;
-
-      auto outerLoop = b.create<AffineForOp>(loc, 0, outerLoopUpperBound);
-      auto outerLoopb = ConversionPatternRewriter::atBlockBegin(
-          outerLoop.getBody(), b.getListener());
-      auto outerLoopiv = outerLoop.getInductionVar();
-
-      auto loadSingleKPack = [&](Value regOffset, Type elementType,
-                                 Value matrix) {
-        Value regIdx = outerLoopb.create<AddIOp>(loc, regOffset, outerLoopiv);
-        Value element = outerLoopb.create<memref::LoadOp>(
-            loc, elementType, matrix, ValueRange{regIdx});
-        return element;
-      };
-      Value matrixAElement =
-          loadSingleKPack(regAConstantOp, matrixAElementType, matrixA);
-      Value matrixBElement =
-          loadSingleKPack(regBConstantOp, matrixBElementType, matrixB);
-
-      auto innerLoop =
-          outerLoopb.create<AffineForOp>(loc, 0, KRepeats * k_base, k_base);
-      auto innerLoopb = ConversionPatternRewriter::atBlockBegin(
-          innerLoop.getBody(), outerLoopb.getListener());
-      auto innerLoopiv = innerLoop.getInductionVar();
-
-      // At this point, we are guaranteed that buffer element vectorization
-      // length (kPack) must be a multiple of k_base. Use extractsliceop
-      // to handle a independent data slice at a time.
-      Value argA = innerLoopb.create<ExtractSliceOp>(
-          loc, argType, matrixAElement, innerLoopiv);
-      Value argB = innerLoopb.create<ExtractSliceOp>(
-          loc, argType, matrixBElement, innerLoopiv);
-      populateMfma(innerLoopb, argA, argB);
-    }
+    };
+
+    auto gemmLoop = b.create<TransformingForOp>(
+        loc, ArrayRef<ValueRange>{startCoords, startCoords, startCoords},
+        ArrayRef<Attribute>{aTransforms, bTransforms, cTransforms}, dimensions,
+        /*strides=*/ArrayRef<int64_t>{1, 1, KPerThread, nResultVectors},
+        /*forceUnroll=*/true, /*useIndexDiffs=*/false);
+    {
+      Value regAOffset = gemmLoop.getLowerCoords(/*domain=*/0)[0];
```

Comment:
Could we maybe go for `regAIdx` or `regAIndex` and so on throughout? Offset makes it sound like we're going to be doing another layer of indexing on top of that

Reply:
Oops sorry I think I didn't see this comment while working on it, otherwise this'd be already included in latest commit. But judging that this is a naming change I'll skip it for now.